modules/lang-painless/src/main/resources/org/elasticsearch/painless/spi/org.elasticsearch.txt

```diff
@@ -175,3 +175,13 @@ class org.elasticsearch.index.similarity.ScriptedSimilarity$Doc {
   int getLength()
   float getFreq()
 }
+
+class org.elasticsearch.script.AnalysisPredicateScript$Term {
```

Comment:
should it be called Token rather than Term? It might be just me but I feel like it better carries the information that there are other attributes here like positions and offsets.

Reply:
I think we should look at creating a new context and adding this to the whitelist of that context. See `plugins/example/painless-whitelist' for some of that.